### PR TITLE
LPS-86937 SF

### PR DESCRIPTION
**Pull request by jbalsas**

### CODE REVIEW

**Comment by IstvanD** [comment]
Hey @topolik,

I was wondering if you could find the time to review the pull request.
Please let me know if you need any help or explanation.

Thank you and best regards,
István

**Comment by topolik** [comment]
Hi,

I wonder why it is a bug? Did Google announce end of life for v2? It looks
more like a feature.

Also, is the solution backwards compatible, ie. would customers need to
register a new api keys or it's a seamless upgrade?

Thanks.


st 14. 11. 2018 12:19 odesílatel István András Dézsi <
notifications@github.com> napsal:

**Comment by IstvanD** [comment]
Hi @topolik 

TL;DR
It may be considered a feature since it introduces the functionally different reCaptcha v3, but it can be also viewed as a bug fix, since it solves an issue with reCatpcha v2 on IE.

------------------------------------------------------------------------------------------------------------------

Introducing reCaptcha v3 causes to a bug with v2 to disappear.
On of our customers reported that when they are using reCapcha with Forms on IE11, the reCaptcha widget loads for some time, then returns to it's base state without validation.
This is a problem that is not Liferay specific and has been already reported:
https://stackoverflow.com/questions/47889701/google-recaptcha-spins-indefinitely-in-ie-11-on-refresh

I found out that the cause of the issue is that the reCaptcha widget needs some time to initialize in the background before if gets rendered. If the page takes 2-3 or more seconds to render, like on a first load, it will have enough time to initialize properly.
But when the page is rendered fast, like under 1 second, typically after a reloading, the reCaptcha widget won't have enough time to initialize before it is load on the page.
If the "I'm not a robot" checkbox is marked, the widget won't be able to establish connection to Google's servers. Please remember that this happens only on IE. Other browsers don't have this timing problem.

I could workaround this problem by introducing a 3 second delay on IE before the widget gets loaded.
https://issues.liferay.com/browse/LPS-86662

But I'm not sure that this would be a proper solution since the user now experiences a 3 second delay before reCaptcha appears.

I discovered that reCaptcha v3 eliminates this problem on IE, since it doesn't requires user interaction (eg. checking the checkbox). Therefore, it is an edge case of a new feature and a bug fix.
It introduces reCaptcha v3 as a new feature, but at the same time, it solves the issue on IE11.

Please let me know what you think of this. I am open to reevaluate this as a feature request, if that is more suitable for this change.

Thank you and best regards,
István

**Comment by topolik** [comment]
I see, thanks for the explanation.

is the solution backwards compatible, ie. would customers need to register
a new api keys or it's a seamless upgrade?

st 14. 11. 2018 13:44 odesílatel István András Dézsi <
notifications@github.com> napsal:

**Comment by topolik** [comment]
Hi @IstvanD,

to backport it we need some non-breaking fix. I'm fine pushing this to master/7.2 but only as a new feature/improvement. 

To bug fix IE 11 I think it's a good idea to go forward with your solution for https://issues.liferay.com/browse/LPS-86662, then it would be a separate fix for IE for 7.1/7.0/... that won't break existing customers env.

Agree?

Thanks.

**Comment by IstvanD** [comment]
Hi @topolik 

I agree with this course.

Thank you for your feedback.

Best regards,
István
